packages/core/src/controllers/LionWebBulkApi.ts

```diff
@@ -54,7 +54,13 @@ class LionWebBulkApiImpl implements LionWebBulkApi {
     async retrieve(req: Request, res: Response): Promise<void> {
         console.log("LionWebBulkApiImpl.retrieve: ")
         const mode = req.query["mode"] as string
-        const depthLimit = Number.parseInt(req.query["depthLimit"] as string)
+        const depthParam = req.query["depthLimit"];
+        let depthLimit: number
```

Comment:
`const depthLimit = (typeof depthParam === "string) ? Number.parseInt(depthParam) : Number.MAX_SAFE_INTEGER`? And: `Number.parseInt` returning `NaN` is not handled now.

Reply:
NaN is handled at line 66.

Reply:
Updated code as described.